src/ch/ch_monitor.c

```diff
@@ -964,3 +973,63 @@ virCHMonitorGetThreadInfo(virCHMonitorPtr mon, bool refresh,
 
     return nthreads;
 }
+
+/**
+ * virCHMonitorGetIOThreads:
+ * @mon: Pointer to the monitor
+ * @iothreads: Location to return array of IOThreadInfo data
+ *
+ * Retrieve the list of iothreads defined/running for the machine
+ *
+ * Returns count of IOThreadInfo structures on success
+ *        -1 on error.
+ */
+int virCHMonitorGetIOThreads(virCHMonitorPtr mon,
+                            virDomainIOThreadInfoPtr **iothreads)
+{
+    size_t nthreads = 0, niothreads=0;
+    int i;
+    virDomainIOThreadInfoPtr *iothreadinfolist = NULL, iothreadinfo = NULL;
+
+    *iothreads = NULL;
+    nthreads = virCHMonitorRefreshThreadInfo(mon);
+
+    if (VIR_ALLOC_N(iothreadinfolist, nthreads) < 0)
+        goto cleanup;
+
+    for (i = 0; i < nthreads; i++){
+        virBitmapPtr map = NULL;
+        if (mon->threads[i].type == virCHThreadTypeIO) {
+            if(VIR_ALLOC(iothreadinfo) < 0)
+                goto cleanup;
+
+            iothreadinfo->iothread_id = mon->threads[i].ioInfo.tid;
+
+            if (!(map = virProcessGetAffinity(iothreadinfo->iothread_id)))
+                goto cleanup;
```

Comment:
This seems like a potential leak since the iothreadinfo would be allocated but the cleanup doesn't free it.

Reply:
Nice Catch! Fixed it in latest push